### PR TITLE
Add support for changes to MSRIOGroup to fix ioctl write

### PR DESCRIPTION
**Pull request by cmcantalupo**

### CODE REVIEW

File: src/MSRIOImp.hpp [hunk 1]
```diff
@@ -34,16 +34,19 @@
...
 namespace geopm
 {
+    class MSRPath;
+
     class MSRIOImp : public MSRIO
     {
         public:
             MSRIOImp();
-            MSRIOImp(int num_cpu);
+            MSRIOImp(int num_cpu, std::shared_ptr<MSRPath>);
```
Comment (cmcantalupo) [inline src/MSRIOImp.hpp]: give the MSRPath parameter a name

File: src/MSRIO.cpp [hunk 2]
```diff
@@ -275,10 +419,9 @@ namespace geopm
     void MSRIOImp::write_batch(const std::vector<uint64_t> &raw_value)
     {
         if (raw_value.size() < m_write_batch.numops) {
-            throw Exception("MSRIOImp::write_batch(): input vector smaller than configured number of operations",
+            throw Exception("MSRIOImp::write_batch(): input vector smaller than configured number of ops",
                             GEOPM_ERROR_INVALID, __FILE__, __LINE__);
         }
-        open_msr_batch();
 #ifdef GEOPM_MSRSAFE_IOCTL_WRITE
```
Comment (cmcantalupo) [inline src/MSRIO.cpp:425]: This will not compile

**Review by dianarg — DISMISSED** [review]
+1 Chris' changes

File: src/MSRFieldControl.cpp [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        double value_inferred = 0.0;
+        uint64_t float_y, float_z;
+        switch (m_function) {
+            case MSR::M_FUNCTION_SCALE:
+                result = (uint64_t)(m_inverse * value);
+                break;
+            case MSR::M_FUNCTION_LOG_HALF:
+                // F = S * 2.0 ^ -X =>
+                // X = log2(S / F)
+                result = (uint64_t)std::log2(m_scalar / value);
```
Comment (cmcantalupo) [inline src/MSRFieldControl.cpp:97]: result = (uint64_t)(-1.0*std::log2(m_inverse * value));

**Review by cmcantalupo — COMMENTED** [review]
Just a nit that I think @dianarg brought up about the log2 function

File: src/MSRFieldControl.cpp (hunk 3 above)
Comment (dianarg) [reply src/MSRFieldControl.cpp:97]: will address this in the next patch adding save/control support

**Review by dianarg — APPROVED** [review]

**Pull request merged**